android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -1074,21 +1074,19 @@ public void onPermissionRequest(final PermissionRequest request) {
       grantedPermissions = new ArrayList<>();
 
       ArrayList<String> requestedAndroidPermissions = new ArrayList<>();
-      String originUrl = request.getOrigin().toString();
-      String originHost;
-
-      try {
-        URL url = new URL(originUrl);
-        originHost = url.getHost();
-      } catch (MalformedURLException e) {
-        request.deny();
-        return;
+      final Uri originUri = request.getOrigin();
+      final String scheme = originUri.getScheme();
+      final int port = originUri.getPort();
+      String origin = scheme + "://" + port;
```

Comment:
It should be host instead of port here.

Reply:
Ops, the initial code was host (getHost) and tested it with that and it works. It got changed in my final commit when I was refactoring🤦‍♂️